R/substitution.R

```diff
@@ -0,0 +1,609 @@
+#'
+#' @description
+#' Wherever there is numerical data that are very small in value, replacement
+#' text may be better for explanatory purposes. The `sub_small_vals()` function
+#' allows for this replacement through specification of a `threshold`, a
+#' `small_pattern`, and the sign of the values to be considered.
+#'
+#' @details
+#' Targeting of values is done through `columns` and additionally by `rows` (if
+#' nothing is provided for `rows` then entire columns are selected). Conditional
+#' formatting is possible by providing a conditional expression to the `rows`
+#' argument. See the Arguments section for more information on this.
+#'
+#' @inheritParams fmt_number
+#' @param threshold The threshold value with which values should be considered
+#'   small enough for replacement.
+#' @param small_pattern The pattern text to be used in place of the suitably
+#'   small values in the rendered table.
+#' @param sign The sign of the numbers to be considered in the replacement. By
+#'   default, we only consider positive values (`"+"`). The other option (`"-"`)
+#'   can be used to consider only negative values.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @section Examples:
+#'
+#' Let's generate a simple, single-column tibble that contains an assortment of
+#' values that could potentially undergo some substitution.
+#'
+#' ```{r}
+#' tbl <- dplyr::tibble(num = c(10^(-4:2), 0, NA))
+#'
+#' tbl
+#' ```
+#'
+#' The `tbl` contains a variety of smaller numbers and some might be small
+#' enough to reformat with a threshold value. With `sub_small_vals()` we can
+#' do just that:
+#'
+#' ```r
+#' tbl %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_small_vals()
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_small_vals_1.png")`
+#' }}
+#'
+#' Small and negative values can also be handled but they are handled specially
+#' by the `sign` parameter. Setting that to `"-"` will format only the small,
+#' negative values.
+#'
+#' ```r
+#' tbl %>%
+#'   dplyr::mutate(num = -num) %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_small_vals(sign = "-")
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_small_vals_2.png")`
+#' }}
+#'
+#' You don't have to settle with the default `threshold` value or the default
+#' replacement pattern (in `small_pattern`). This can be changed and the
+#' `"{x}"` in `small_pattern` (which uses the `threshold` value) can even be
+#' omitted.
+#'
+#' ```r
+#' tbl %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_small_vals(
+#'     threshold = 0.0005,
+#'     small_pattern = "smol"
+#'   )
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_small_vals_3.png")`
+#' }}
+#'
+#' @family Format Data
+#' @section Function ID:
+#' 3-18
+#'
+#' @import rlang
+#' @export
+sub_small_vals <- function(
+    data,
+    columns = everything(),
+    rows = everything(),
+    threshold = 0.01,
+    small_pattern = "<{x}",
```

Comment:
```suggestion
    small_pattern = if (sign == "+") "<{x}" else md("<*abs*({x})"),
```

Reply:
I used this minor variation: `if (sign == "+") "<{x}" else md("<*abs*(-{x})")`